src/test/mempoolestimator_tests.cpp

```diff
@@ -0,0 +1,113 @@
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <policy/feerate.h>
+#include <policy/forecaster.h>
+#include <policy/mempool_fees.h>
+#include <test/util/random.h>
+#include <test/util/txmempool.h>
+#include <txmempool.h>
+#include <uint256.h>
+#include <util/strencodings.h>
+#include <validation.h>
+
+
+#include <test/util/setup_common.h>
+
+#include <memory>
+#include <string>
+
+#include <boost/test/unit_test.hpp>
+
+BOOST_FIXTURE_TEST_SUITE(mempoolestimator_tests, TestChain100Setup)
+
+static inline CTransactionRef make_random_tx()
+{
+    CMutableTransaction tx = CMutableTransaction();
+    tx.vin.resize(1);
+    tx.vout.resize(1);
+    tx.vin[0].prevout.hash = Txid::FromUint256(InsecureRand256());
+    tx.vin[0].prevout.n = 0;
+    tx.vin[0].scriptSig << OP_TRUE;
+    tx.vout[0].scriptPubKey = CScript() << OP_11 << OP_EQUAL;
+    tx.vout[0].nValue = COIN;
+    return MakeTransactionRef(tx);
+}
+
+BOOST_AUTO_TEST_CASE(MempoolEstimator)
+{
+    m_node.mempool.reset();
+    m_node.mempool = std::make_unique<CTxMemPool>(MemPoolOptionsForTest(m_node));
+    auto mempool_fee_estimator = std::make_unique<MemPoolPolicyEstimator>(m_node.mempool.get(), &(m_node.chainman->ActiveChainstate()));
+
+    LOCK(cs_main);
+    LOCK(m_node.mempool->cs);
+
+    {
+        // Test when targetBlocks > MEMPOOL_FORECAST_MAX_TARGET
+        const auto fee_estimate = mempool_fee_estimator->EstimateFee(MEMPOOL_FORECAST_MAX_TARGET + 1);
+        BOOST_CHECK(fee_estimate.empty() == true);
+        BOOST_CHECK(fee_estimate.m_err_message == strprintf("%s: Confirmation target %s is above maximum limit of %s, mempool conditions might change and forecast above %s block target are unreliable",
+                                                            MEMPOOL_FORECAST_NAME_STR, MEMPOOL_FORECAST_MAX_TARGET + 1, MEMPOOL_FORECAST_MAX_TARGET, MEMPOOL_FORECAST_MAX_TARGET));
+    }
+
+    {
+        // Test ehrn mempool is not loaded
+        const auto fee_estimate = mempool_fee_estimator->EstimateFee(MEMPOOL_FORECAST_MAX_TARGET);
+        BOOST_CHECK(fee_estimate.empty() == true);
+        BOOST_CHECK(fee_estimate.m_err_message == strprintf("%s: Mempool did not finished loading, can't get accurate fee rate forecast", MEMPOOL_FORECAST_NAME_STR));
+    }
+
+    {
+        // Test when no transactions available in the mempool yet
+        m_node.mempool->SetLoadTried(true);
+        const auto fee_estimate = mempool_fee_estimator->EstimateFee(MEMPOOL_FORECAST_MAX_TARGET);
+        BOOST_CHECK(fee_estimate.empty() == true);
+        BOOST_CHECK(fee_estimate.m_err_message == strprintf("%s: No transactions available in the mempool", MEMPOOL_FORECAST_NAME_STR));
+    }
+    BOOST_CHECK(m_node.mempool->GetTotalTxSize() == 0);
+    TestMemPoolEntryHelper entry;
+
+    const CAmount low_fee{CENT / 3000};
+    const CAmount med_fee{CENT / 100};
+    const CAmount high_fee{CENT / 10};
+
+    // Test when mempool transactions are not enough
```

Comment:
In  [test]: add MemPoolPolicyEstimator unit test:

```suggestion
    // Test when there are not enough mempool transactions to get an accurate estimate
````